metadata-io/src/main/java/com/linkedin/metadata/timeseries/elastic/ElasticSearchTimeseriesAspectService.java

```diff
@@ -175,4 +186,56 @@ public GenericTable getAggregatedStats(@Nonnull String entityName, @Nonnull Stri
       @Nullable GroupingBucket[] groupingBuckets) {
     return _esAggregatedStatsDAO.getAggregatedStats(entityName, aspectName, aggregationSpecs, filter, groupingBuckets);
   }
+
+  /**
+   * A generic delete by filter API which uses elasticsearch's deleteByQuery.
+   * NOTE: There is no need for the client to explicitly walk each scroll page with this approach. Elastic will synchronously
+   * delete all of the documents matching the query that is specified by the filter, and internally handles the batching logic
+   * by the scroll page size specified(i.e. the DEFAULT_LIMIT value of 10,000).
+   * @param entityName the name of the entity.
+   * @param aspectName the name of the aspect.
+   * @param filter the filter to be used for deletion of the documents on the index.
+   * @return the numer of documents returned.
+   */
+  @Nonnull
+  @Override
+  public DeleteAspectValuesResult deleteAspectValues(@Nonnull String entityName, @Nonnull String aspectName,
+      @Nonnull Filter filter) {
+    final String indexName = _indexConvention.getTimeseriesAspectIndexName(entityName, aspectName);
+    final BoolQueryBuilder filterQueryBuilder = QueryBuilders.boolQuery().must(ESUtils.buildFilterQuery(filter));
+    final DeleteByQueryRequest deleteByQueryRequest = new DeleteByQueryRequest(indexName).setQuery(filterQueryBuilder)
+        .setBatchSize(DEFAULT_LIMIT)
+        .setRefresh(true)
+        .setTimeout(TimeValue.timeValueMinutes(10));
+    try {
+      final BulkByScrollResponse response = _searchClient.deleteByQuery(deleteByQueryRequest, RequestOptions.DEFAULT);
```

Comment:
does this delete over 10K documents (es limit for read side)?

Reply:
just making sure

Reply:
Yes, it does it in batches of 10K documents underneath via multiple delete queries.